pkg/helpers/node.go

```diff
@@ -0,0 +1,76 @@
+package helpers
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+
+	"github.com/devbuddy/devbuddy/pkg/config"
+	"github.com/devbuddy/devbuddy/pkg/executor"
+	"github.com/devbuddy/devbuddy/pkg/utils"
+)
+
+type Node struct {
+	version string
+	path    string
+	tarDir  string
+}
+
+func NewNode(cfg *config.Config, version string) *Node {
+	return &Node{
+		version: version,
+		path:    cfg.DataDir("node", version),
+		tarDir:  cfg.DataDir("node"),
+	}
+}
+
+func (n *Node) Exists() bool {
+	return utils.PathExists(n.Which("node"))
+}
+
+func (n *Node) Path() string {
+	return n.path
+}
+
+func (n *Node) BinPath() string {
+	return path.Join(n.path, "bin")
+}
+
+func (n *Node) Which(program string) string {
+	return path.Join(n.path, "bin", program)
+}
+
+func (n *Node) Install() error {
+	if runtime.GOARCH != "amd64" {
+		return fmt.Errorf("Binary distribution are not available for %s architecture", runtime.GOARCH)
```

Comment:
Not sure it's right since most architectures are available. maybe we should rephrase this to something like:
`devbuddy is not currently supporting installation of node on %s architecture` 

?